realm/src/androidTest/java/io/realm/RealmListTest.java

```diff
@@ -461,6 +473,186 @@ public void testRemoveByObject() {
         assertEquals(TEST_OBJECTS - 1, dogs.size());
     }
 
+    public void testAddAtAfterContainerObjectRemoved() {
+        RealmList<Dog> dogs = createDeletedRealmList();
+
+        testRealm.beginTransaction();
+        try {
+            Dog dog = testRealm.createObject(Dog.class);
+            dog.setName("Dog");
+            try {
+                dogs.add(0, dog);
+                fail("accessing removed object must throw IllegalStatException.");
```

Comment:
No need to add a fail message if the intent is clear from the surrounding code. In this case, you can see the IllegalStateException being caught just below.